internal/testutils/xds/e2e/setup_management_server.go

```diff
@@ -110,3 +111,30 @@ func DefaultBootstrapContents(nodeID, serverURI string) ([]byte, error) {
 	}
 	return bs, nil
 }
+
+// SetupDNS unregisters the DNS resolver and registers a manual resolver for the
```

Comment:
Hmm, interesting. Instead of using the passthrough resolver, did you consider mocking the results for the DNS name of the xDS server instead? To me, what you ran into works as intended...

Another thing that may be useful to the facility to mock DNS is to pass local addresses through, which may also have solved your problem.

Reply:
My point was that from a testing perspective, if you're going to intercept all DNS calls, then it makes a lot of sense to have to do it for the management server as well...